tests/test_ssa_generation.py

```diff
@@ -1077,3 +1079,28 @@ def test_issue_1748():
         operations = f.slithir_operations
         assign_op = operations[0]
         assert isinstance(assign_op, InitArray)
+
+
+def test_issue_1776():
+    source = """
+    contract Contract {
+        function foo() public returns (uint) {
+            uint[5][10][] memory arr = new uint[5][10][](2);
+            return 0;
+        }
+    }
+    """
+    with slither_from_source(source) as slither:
+        c = slither.get_contract_from_name("Contract")[0]
+        f = c.functions[0]
+        operations = f.slithir_operations
+        new_op = operations[0]
+        lvalue = new_op.lvalue
+        lvalue_type = lvalue.type
+        assert isinstance(lvalue_type, ArrayType)
+        lvalue_type1 = lvalue_type.type
+        assert isinstance(lvalue_type1, ArrayType)
+        assert lvalue_type1.length_value.value == "10"
+        lvalue_type2 = lvalue_type1.type
+        assert isinstance(lvalue_type2, ArrayType)
+        assert lvalue_type2.length_value.value == "5"
```

Comment:
This seems backwards?

Reply:
What do you mean `backwards`? 
Here, I am checking the nested array type and length should be parsed and converted to IR correctly.

Reply:
Nvm, I misunderstood which length value corresponded to which type. This seems correct